tests/EvmAcceptanceTests/test/ContractDestruction.js

```diff
@@ -1,13 +1,50 @@
 const {expect} = require("chai");
+const {web3} = require("hardhat");
+const web3_helper = require("../helper/Web3Helper");
+const general_helper = require("../helper/GeneralHelper");
 
 describe("Contract destruction with web3.js", function () {
-  describe("via user method call", function () {
-    // TODO: Consider adding the test case when the receiving address does not exist.
-    it("should be destructed and coins in the contract should be transferred to the address specified in the method");
+  let contract;
+  const gasLimit = "750000";
+  const amountPaid = web3.utils.toBN(web3.utils.toWei("300", "gwei"));
+  let options;
+  before(async function () {
+    contract = await web3_helper.deploy("ParentContract", {gasLimit, value: amountPaid});
+    options = await web3_helper.getCommonOptions();
   });
-  describe("through a method call by another contract", function () {
-    // TODO: Consider adding the test case when the receiving address does not exist.
-    it("should be destructed and coins in the contract should be transferred to the address specified in the method");
+
+  describe("When a user method call", function () {
+    it("should be destructed and coins in the contract should be transferred to the address specified in the method", async function () {
+      expect(await contract.methods.getPaidValue().call(options)).to.be.eq(amountPaid);
+      const destAccount = await web3.eth.accounts.privateKeyToAccount(general_helper.getPrivateAddressAt(1)).address;
+      const prevBalance = await web3.eth.getBalance(destAccount);
+      expect(
+        await contract.methods
+          .returnToSenderAndDestruct(destAccount)
+          .send({gasLimit: 1000000, from: web3_helper.getPrimaryAccountAddress()})
+      ).to.be.not.null;
+      const newBalance = await web3.eth.getBalance(destAccount);
+      // Dest Account should have prevBalance + amountPaid
+      expect(web3.utils.toBN(newBalance)).to.be.equal(web3.utils.toBN(prevBalance).add(amountPaid));
+    });
+  });
+
+  describe("When a method call happens through another contract", function () {
```

Comment:
one general note for destruction. isn't there any way to test if the contract does not exist anymore after the destruction?

Reply:
So far it's not, we have to check how it's done in other networks and decide what would be suitable for our blockchain (either empty account or non-existing one).